oshdb/src/test/java/org/heigit/ohsome/oshdb/contribution/ContributionsWayTest.java

```diff
@@ -0,0 +1,364 @@
+        osh(1,
+            node(2, 2000, 202, 2, tags(), 0, 0),
+            node(1, 1000, 101, 1, tags(), 0, 0)));
+
+    var versions = ways(1,
+        way(1, 1000, 101, 1, tags(), mems(1)));
+    var osh = osh(versions, nodes);
+    var contribs = Contributions.of(osh, 1500);
+
+    assertTrue(contribs.hasNext());
+    var contrib = contribs.next();
+    assertContrib(1000, 101, 1, versions.get(0), contrib);
+    assertEquals(EnumSet.of(CREATION), contrib.getTypes());
+
+    assertFalse(contribs.hasNext());
+  }
+
+  @Test
+  void testNewMemberContribOutside() {
+    var nodes = List.of(
+        osh(1,
+            node(1, 1000, 101, 1, tags(), 0, 0)),
+        osh(2,
+            node(2, 3000, 303, 3, tags(), 0, 0),
+            node(1, 1000, 101, 1, tags(), 0, 0)));
+
+    var versions = ways(1,
+        way(2, 2000, 202, 2, tags(), mems(1)),
+        way(1, 1000, 101, 1, tags(), mems(2)));
+    var osh = osh(versions, nodes);
+    var contribs = Contributions.of(osh);
+
+    assertTrue(contribs.hasNext());
+    var contrib = contribs.next();
+    assertContrib(2000, 202, 2, versions.get(0), contrib);
+    assertEquals(EnumSet.of(MEMBER_CHANGE), contrib.getTypes());
+
+    assertTrue(contribs.hasNext());
+    contrib = contribs.next();
+    assertContrib(1000, 101, 1, versions.get(1), contrib);
+    assertEquals(EnumSet.of(CREATION), contrib.getTypes());
+
+    assertFalse(contribs.hasNext());
+  }
+
+  @Test
+  void testNewMemberNoContribInside() {
+    var nodes = List.of(
+        osh(1,
+            node(1, 1000, 101, 1, tags(), 0, 0)),
+        osh(2,
+            node(1, 3000, 303, 3, tags(), 0, 0)));
+
+    var versions = ways(1,
+        way(2, 2000, 202, 2, tags(), mems(1)),
+        way(1, 1000, 101, 1, tags(), mems(2)));
+    var osh = osh(versions, nodes);
+    var contribs = Contributions.of(osh);
+
+    assertTrue(contribs.hasNext());
+    var contrib = contribs.next();
+    assertContrib(2000, 202, 2, versions.get(0), contrib);
+    assertEquals(EnumSet.of(MEMBER_CHANGE), contrib.getTypes());
+
+    assertTrue(contribs.hasNext());
+    contrib = contribs.next();
+    assertContrib(1000, 101, 1, versions.get(1), contrib);
+    assertEquals(EnumSet.of(CREATION), contrib.getTypes());
+
+    assertFalse(contribs.hasNext());
+  }
+
+  @Test
+  void testDifferentMember() {
+    var nodes = List.of(
+        osh(1,
+            node(1, 1000, 101, 1, tags(), 0, 0)),
+        osh(2,
+            node(1, 1000, 101, 1, tags(), 0, 0)),
+        osh(3,
+            node(1, 1000, 101, 1, tags(), 0, 0)));
+
+    var versions = ways(1,
+        way(2, 2000, 202, 2, tags(), mems(2, 3, 2)),
+        way(1, 1000, 101, 1, tags(), mems(1, 2)));
+    var osh = osh(versions, nodes);
+    var contribs = Contributions.of(osh);
+
+    assertTrue(contribs.hasNext());
+    var contrib = contribs.next();
+    assertContrib(2000, 202, 2, versions.get(0), contrib);
+    assertEquals(EnumSet.of(MEMBER_CHANGE), contrib.getTypes());
```

Comment:
Should we check if the members are set correctly, or what do you want to check here (compared to other tests)?